src/sectionproperties/pre/library/timber_sections.py

```diff
@@ -0,0 +1,54 @@
+"""Timber sections library."""
+
+from __future__ import annotations
+
+import sectionproperties.pre.geometry as geometry
+import sectionproperties.pre.library.primitive_sections as primitive_sections
+import sectionproperties.pre.pre as pre
+
+
+def clt_rectangular_section(
+    d: list[float], lay_orient: list[pre.Material], b: float
+) -> geometry.CompoundGeometry:
+    """Constructs a timber rectangular section.
+
+    Constructs a timber rectangular section of depth ``d`` and width ``b``.
+
+    .. note::
+
+    Args:
+        d: Timber layer section thickness
+        lay_orient: A list of materials for each layer defined by the user.
```

Comment:
It may be worth adding that layers are added from top to bottom in this implementation (users may expect a bottom up by default).